services/supervisor/src/api.rs

```diff
@@ -12,6 +12,578 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+//! HTTP API for the supervisor service. Supervisor API is divided into public and private
+//! parts, with public part intended for unauthorized use, and private parts intended to be
+//! used by network administrator for the Exonum blockchain configuration.
+//!
+//! # Table of Contents
+//!
+//! - Public API:
+//!
+//!     - [Obtaining consensus configuration](#obtaining-consensus-configuration)
+//!     - [Obtaining pending configuration proposal](#obtaining-pending-configuration-proposal)
+//!
+//! - Private API:
+//!
+//!     - [Request to deploy an artifact](#request-to-deploy-an-artifact)
+//!     - [Request service migration](#request-service-migration)
+//!     - [Request to accept new configuration](#request-to-accept-new-configuration)
+//!     - [Vote for configuration proposal](#vote-for-configuration-proposal)
+//!     - [Obtain current configuration number](#obtain-current-configuration-number)
+//!     - [Obtain current supervisor operating mode](#obtain-current-supervisor-operating-mode)
+//!     - [Check the deployment status](#check-deployment-status)
+//!     - [Check the migration status](#check-migration-status)
+//!
+//!
+//! # Public API
+//!
+//! ## Obtaining Consensus Configuration
+//!
+//! | Property    | Value |
+//! |-------------|-------|
+//! | Path        | `/api/supervisor/v1/consensus-config` |
+//! | Method      | GET   |
+//! | Query type  | - |
+//! | Return type | [`BlockInfo`] |
+//!
+//! Returns the current consensus configuration.
+//!
+//! [`ConsensusConfig`]: https://docs.rs/exonum/latest/exonum/blockchain/config/struct.ConsensusConfig.html
+//!
+//! ```
+//! use exonum::blockchain::ConsensusConfig;
+//! use exonum_rust_runtime::ServiceFactory;
+//! use exonum_supervisor::Supervisor;
+//! use exonum_testkit::{ApiKind, TestKitBuilder};
+//!
+//! # fn main() -> Result<(), failure::Error> {
+//! let mut testkit = TestKitBuilder::validator()
+//!     .with_rust_service(Supervisor)
+//!     .with_artifact(Supervisor.artifact_id())
+//!     .with_instance(Supervisor::simple())
+//!     .create();
+//!
+//! let consensus_config: ConsensusConfig = testkit
+//!     .api()
+//!     .public(ApiKind::Service("supervisor"))
+//!     .get("consensus-config")?;
+//!
+//! # Ok(())
+//! # }
+//! ```
+//!
+//! ## Obtaining Pending Configuration Proposal
+//!
+//! | Property    | Value |
+//! |-------------|-------|
+//! | Path        | `/api/supervisor/v1/config-proposal` |
+//! | Method      | GET   |
+//! | Query type  | - |
+//! | Return type | `Option<[ConfigProposalWithHash]>` |
+//!
+//! Returns the configuration proposal which is currently pending. Returns `None` if there is no
+//! pending configuration at the moment.
+//!
+//! [ConfigProposalWithHash]: ../struct.ConfigProposalWithHash.html
+//!
+//! ```
+//! use exonum_rust_runtime::ServiceFactory;
```

Comment:
Nit: IMO, it would be safe to skip imports / `testkit` initialization if it coincides with the earlier tests:

```rust
let mut testkit = // same as in previous example
let pending_proposal: Option<ConfigProposalWithHash> = testkit
```

Reply:
Is it OK now, or should I hide something else as well?